homeassistant/components/powerwall/binary_sensor.py

```diff
@@ -0,0 +1,183 @@
+"""Support for August sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_CONNECTIVITY,
+    BinarySensorDevice,
+)
+from homeassistant.const import DEVICE_CLASS_POWER
+
+from . import PowerWallEntity
+from .const import (
+    ATTR_GRID_CODE,
+    ATTR_NOMINAL_SYSTEM_POWER,
+    ATTR_REGION,
+    DOMAIN,
+    POWERWALL_API_GRID_STATUS,
+    POWERWALL_API_SITEMASTER,
+    POWERWALL_CONNECTED_KEY,
+    POWERWALL_COORDINATOR,
+    POWERWALL_GRID_CODE,
+    POWERWALL_GRID_ONLINE,
+    POWERWALL_IP_ADDRESS,
+    POWERWALL_NOMINAL_SYSTEM_POWER,
+    POWERWALL_REGION,
+    POWERWALL_RUNNING_KEY,
+    POWERWALL_SITE_INFO,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the August sensors."""
+    powerwall_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    coordinator = powerwall_data[POWERWALL_COORDINATOR]
+    site_info = powerwall_data[POWERWALL_SITE_INFO]
+    ip_address = powerwall_data[POWERWALL_IP_ADDRESS]
+
+    entities = []
+    entities.append(PowerWallRunningSensor(coordinator, site_info, ip_address))
+    entities.append(PowerWallGridStatusSensor(coordinator, site_info, ip_address))
+    entities.append(PowerWallConnectedSensor(coordinator, site_info, ip_address))
```

Comment:
Maybe consolidate? It's one more line, but more dry. :man_shrugging: 
```suggestion
    for sensor_class in (
        PowerWallRunningSensor, PowerWallGridStatusSensor, PowerWallConnectedSensor
    ):
        entities.append(sensor_class(coordinator, site_info, ip_address))
```